Dis/enable beam centre component boxes on component update

TTitcombe opened the pull request:
**Description of work.**
This PR automatically updates the HAB/LAB component checkboxes in the ISIS SANS beam centre finder tab if the reduction mode has been changed.

If HAB mode has been selected, the beam centre finder LAB checkbox is unchecked and disabled, and vice versa. This change clarifies which components have been selected for reduction.

**Report to:** Steve King ISIS

**To test:**
1. Interfaces -> SANS -> ISIS SANS
2. In the settings tab, select the HAB reduction mode ("Hab" or "front")
3. Go to the beam centre finder tab. The HAB/Hab/front checkbox should be enabled and checked. The LAB/main-detector/rear checkbox should be unchecked and disabled
4. Back in settings, select LAB reduction mode ("main-detector" or "rear")
5. The checkboxes should be in the opposite state to step (3)
6. Back in settings, select "merged" or "all" mode
7. Both checkboxes in beam center finder tab should be enabled and checked

Fixes #25997 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

TTitcombe committed:
Dis/enable beam centre component boxes on component update

Add unit tests

Update sans 4.1 release notes

re #25997

hsautessella commented on scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py:
```diff
@@ -878,6 +882,17 @@ def on_save_other(self):
+            self._beam_centre_presenter.update_all_selected()
+
     # def _validate_rows(self):
     #     """
```
If this commented out method is unused, should it be deleted?

TTitcombe replied:
Git history shows it was commented out at least 11 months ago, so it's probably safe to remove

hsautessella commented on scripts/Interface/ui/sans_isis/beam_centre.py:
```diff
@@ -155,6 +155,18 @@ def set_run_button_to_normal(self):
+            self.update_hab_check_box.setChecked(True)
+        else:
+            self.update_hab_check_box.setChecked(False)
+            self.update_hab_check_box.setEnabled(False)
```
If `enabled` is True or False can this method be condensed to just:
```python
def enable_update_hab(self, enabled):
    self.update_hab_check_box.setEnabled(enabled)
    self.update_hab_check_box.setChecked(enabled)
```

TTitcombe replied:
Yep, this was left over from testing when I thought that the checkboxes could only be checked/unchecked when they were enabled. I'll change this

TTitcombe committed:
Act on reviewer comments

Remove old commented out code in run_tab_presenter

Simplify HAB checkbox enabling method

re #25997

gemmaguest commented:
@TTitcombe Could you resolve the conflicts?

NickDraper committed:
Merge branch 'master' into 25997_incorrect_beam_centre_finder_components
